auth/authenticator.go

```diff
@@ -0,0 +1,82 @@
+package auth
+
+import (
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+
+	"github.com/ivanterekh/go-skeleton/env"
+	"github.com/ivanterekh/go-skeleton/model"
+	"github.com/ivanterekh/go-skeleton/repository/users"
+)
+
+// Authenticator is a service for logging in and
+// authenticating users.
+type Authenticator struct {
+	exp    time.Duration
+	method *jwt.SigningMethodHMAC
+	secret string
+	users  users.Repository
+}
+
+// Exp returns expiry time.
+func (a *Authenticator) Exp() time.Duration {
+	return a.exp
+}
+
+// NewAuthenticator returns a new authenticator instance.
+func NewAuthenticator() *Authenticator {
+	return &Authenticator{
+		exp:    time.Hour * 48,
```

Comment:
Why 48 hours? It's a kind of magic number here. How can we change that? I'd recommend creating another `NewAuthenticator` method with arguments and this one call as `DefaultAuthenticator() {return NewAuthenticator() }`. But it's optional